tests/test_experiments.py

```diff
@@ -14,6 +18,257 @@
+        "running tests relative to the base imitation project folder.",
+    )
+
+EXPERIMENTS_DIR = types.parse_path("experiments")
+COMMANDS_PY_PATH = EXPERIMENTS_DIR / "commands.py"
+
+
+def _get_benchmarking_path(benchmarking_file):
+    return os.path.join(BENCHMARKING_DIR.stem, benchmarking_file)
+
+
+def _run_commands_from_flags(**kwargs) -> List[str]:
+    """Run commands.py with flags derived from the given `kwargs`.
+
+    This is a helper function to reduce boilerplate code in the tests
+    for commands.py.
+
+    Each key-value pair in kwargs corresponds to one flag.
+    If the value in the key-value is True, then the flag has the form "--key".
+    Otherwise, the flag has the form "--key=value".
+
+    For example, _run_commands_from_flags(name="baz", remote=True)
+    will execute the following python command:
+
+    python experiments/commands.py --name=baz --remote
+
+    The function will return the parsed output of executing that command.
+
+    Args:
+        kwargs: keyword arguments from which to derive flags.
+
+    Returns:
+        A list where the ith entry of the list is the ith line printed out
+        by the python command.
+    """
+    flags = []
+
+    # Add some defaults so that most tests can skip specifying these flags.
+    if "name" not in kwargs:
+        flags.append("--name=run0")
+
+    if "cfg_pattern" not in kwargs:
+        cfg_pattern = _get_benchmarking_path("fast_dagger_seals_cartpole.json")
+        flags.append("--cfg_pattern=" + cfg_pattern)
+
+    if "output_dir" not in kwargs:
+        flags.append("--output_dir=output")
+
+    # Build the flags.
+    for key in kwargs:
+        flag = "--" + key
+        if (not isinstance(kwargs[key], bool)) and kwargs[key]:
+            flag += "=" + kwargs[key]
+        flags.append(flag)
+
+    py_command = f"python {COMMANDS_PY_PATH} " + " ".join(flags)
+
+    completed_process = subprocess.run(
+        py_command,
+        shell=True,
+        capture_output=True,
+        stdin=subprocess.DEVNULL,
+        check=True,
+    )
+
+    # Each line of the formatted stdout is a command.
+    formatted_stdout = completed_process.stdout.decode("ascii").strip()
+    commands = formatted_stdout.split("\n")
+
+    return commands
+
+
+def test_commands_local_config():
+    commands = _run_commands_from_flags()
+    assert len(commands) == 1
+    expected = (
+        "python -m imitation.scripts.train_imitation dagger "
+        "--capture=sys --name=run0 "
+        "--file_storage=output/sacred/"
+        "$USER-cmd-run0-dagger-0-8bf911a8 "
+        "with benchmarking/fast_dagger_seals_cartpole.json "
+        "seed=0 logging.log_root=output"
+    )
+    assert commands[0] == expected
+
+
+def test_commands_local_config_runs(tmpdir):
+    output_dir = types.parse_path(tmpdir).as_posix()
+    commands = _run_commands_from_flags(output_dir=output_dir)
+    assert len(commands) == 1
+    expected = (
```

Comment:
same as above but with `output_dir`? Makes sense to split up the tests to more precisely localize failures, but maybe just have one `EXPECTED_LOCAL_CONFIG` or the like template that you can reuse between these two tests to cut down on duplication.